src/eth1/impl/ethers.ts

```diff
@@ -58,55 +68,85 @@ export class EthersEth1Notifier extends EventEmitter implements Eth1Notifier {
   public async processBlockHeadUpdate(blockNumber): Promise<void> {
     logger.debug(`Received eth1 block ${blockNumber}`);
     const block = await this.provider.getBlock(blockNumber);
+    this._latestBlockHash = Buffer.from(block.hash.substr(2), 'hex');
     this.emit('block', block);
   }
 
   public async processDepositLog(dataHex: string, indexHex: string): Promise<void> {
     const dataBuf = Buffer.from(dataHex.substr(2), 'hex');
     const index = Buffer.from(indexHex.substr(2), 'hex').readUIntLE(0, 6);
 
-    logger.info(`Received validator deposit event index=${index}. Current deposit count=${this.depositCount}`);
+    logger.info(
+      `Received validator deposit event index=${index}. Current deposit count=${this.depositCount}`
+    );
     if (index !== this.depositCount) {
       logger.warn(`Validator deposit with index=${index} received out of order.`);
       // deposit processed out of order
       return;
     }
     this.depositCount++;
-
     const data: DepositData = deserialize(dataBuf, DepositData);
-
-    // TODO: Add deposit to merkle trie/db
-    this.emit('deposit', data, index);
+    const deposit: Deposit = {
+      index: index,
+      //TODO: calculate proof
+      proof: [],
+      data
+    };
+    //after genesis stop storing in genesisDeposit bucket
+    if(!this.genesisBlockHash) {
+      await this.db.setGenesisDeposit(deposit);
+    }
+    this.emit('deposit', deposit);
   }
 
-  public async processEth2GenesisLog(depositRootHex: string, depositCountHex: string, timeHex: string, event: ethers.Event): Promise<void> {
+  public async processEth2GenesisLog(
+    depositRootHex: string,
+    depositCountHex: string,
+    timeHex: string,
+    event: ethers.Event
+  ): Promise<void> {
     const depositRoot = Buffer.from(depositRootHex.substr(2), 'hex');
     const depositCount = Buffer.from(depositCountHex.substr(2), 'hex').readUIntLE(0, 6);
     const time = new BN(Buffer.from(timeHex.substr(2), 'hex').readUIntLE(0, 6));
     const blockHash = Buffer.from(event.blockHash.substr(2), 'hex');
+    this.genesisBlockHash = event.blockNumber;
     logger.info(`Received Eth2Genesis event. blockNumber=${event.blockNumber}, time=${time}`);
-    // TODO: Ensure the deposit root is the same that we've stored
 
     const genesisEth1Data: Eth1Data = {
       depositRoot,
       blockHash,
       depositCount,
     };
-
-    this.chainStarted = true;
-    this.emit('eth2genesis', time, await this.genesisDeposits(), genesisEth1Data);
+    const genesisDeposits = await this.genesisDeposits();
+    this.emit('eth2genesis', time, genesisDeposits, genesisEth1Data);
+    //from now on it will be kept in BeaconBlock
+    await this.db.deleteGenesisDeposits(genesisDeposits);
   }
 
   public async genesisDeposits(): Promise<Deposit[]> {
-    return [];
+    return this.db.getGenesisDeposits();
```

Comment:
does this need to be `await`ed?

Reply:
It doesn't matter, it returns promise anyways, there is a test that proves it :)